driver-core/src/main/java/com/datastax/driver/core/PagingState.java

```diff
@@ -0,0 +1,176 @@
+/*
+ *      Copyright (C) 2012-2015 DataStax Inc.
+ *
+ *   Licensed under the Apache License, Version 2.0 (the "License");
+ *   you may not use this file except in compliance with the License.
+ *   You may obtain a copy of the License at
+ *
+ *      http://www.apache.org/licenses/LICENSE-2.0
+ *
+ *   Unless required by applicable law or agreed to in writing, software
+ *   distributed under the License is distributed on an "AS IS" BASIS,
+ *   WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ *   See the License for the specific language governing permissions and
+ *   limitations under the License.
+ */
+
+package com.datastax.driver.core;
+
+import java.nio.ByteBuffer;
+import java.security.MessageDigest;
+import java.security.NoSuchAlgorithmException;
+
+import com.datastax.driver.core.exceptions.PagingStateException;
+import com.datastax.driver.core.utils.Bytes;
+
+/**
+ * The paging state of a query.
+ *
+ * This object represents the next page to be fetched if the query is
+ * multi page. It can be saved and reused later on the same statement.
+ *
+ * The PagingState can be serialized and deserialized either in a String
+ * or in a ByteBuffer.
+ */
+public class PagingState {
+
+    private ByteBuffer pagingState;
```

Comment:
The public API would still expose a single field, but it would be built on-demand from the two internal arrays.
So I think Andy's suggestion makes sense, as it would make the internal code cleaner. Sorry for changing my mind again on this.